Common/cpp/NativeModules/NativeReanimatedModule.cpp

```diff
@@ -173,87 +160,62 @@ NativeReanimatedModule::NativeReanimatedModule(
 
 void NativeReanimatedModule::installCoreFunctions(
     jsi::Runtime &rt,
-    const jsi::Value &valueSetter) {
-  this->valueSetter = ShareableValue::adapt(rt, valueSetter, this);
-}
-
-jsi::Value NativeReanimatedModule::makeShareable(
-    jsi::Runtime &rt,
-    const jsi::Value &value) {
-  return ShareableValue::adapt(rt, value, this)->getValue(rt);
-}
-
-jsi::Value NativeReanimatedModule::makeMutable(
-    jsi::Runtime &rt,
-    const jsi::Value &value) {
-  return ShareableValue::adapt(rt, value, this, ValueType::MutableValueType)
-      ->getValue(rt);
-}
-
-jsi::Value NativeReanimatedModule::makeRemote(
-    jsi::Runtime &rt,
-    const jsi::Value &value) {
-  return ShareableValue::adapt(rt, value, this, ValueType::RemoteObjectType)
-      ->getValue(rt);
+    const jsi::Value &valueUnpacker) {
+  runtimeHelper =
+      std::make_shared<JSRuntimeHelper>(&rt, this->runtime.get(), scheduler);
+  runtimeHelper->valueUnpacker =
+      std::make_shared<CoreFunction>(runtimeHelper.get(), valueUnpacker);
 }
 
-jsi::Value NativeReanimatedModule::startMapper(
+void NativeReanimatedModule::scheduleOnUI(
     jsi::Runtime &rt,
-    const jsi::Value &worklet,
-    const jsi::Value &inputs,
-    const jsi::Value &outputs,
-    const jsi::Value &updater,
-    const jsi::Value &viewDescriptors) {
-  static unsigned long MAPPER_ID = 1;
-
-  unsigned long newMapperId = MAPPER_ID++;
-  auto mapperShareable = ShareableValue::adapt(rt, worklet, this);
-  auto inputMutables =
-      extractMutablesFromArray(rt, inputs.asObject(rt).asArray(rt), this);
-  auto outputMutables =
-      extractMutablesFromArray(rt, outputs.asObject(rt).asArray(rt), this);
-
-  int optimalizationLvl = 0;
-  auto optimalization =
-      updater.asObject(rt).getProperty(rt, "__optimalization");
-  if (optimalization.isNumber()) {
-    optimalizationLvl = optimalization.asNumber();
-  }
-  auto updaterSV = ShareableValue::adapt(rt, updater, this);
-  auto viewDescriptorsSV = ShareableValue::adapt(rt, viewDescriptors, this);
-
+    const jsi::Value &worklet) {
+  auto shareableWorklet = extractShareableOrThrow(rt, worklet);
+  assert(
+      shareableWorklet->valueType == Shareable::WorkletType &&
+      "only worklets can be scheduled to run on UI");
```

Comment:
Is this warning meant to be for internal purposes only? We have this cool warning message "Tried to synchronously call function ..." with further instructions for the library user, is it something similar or totally different?

Reply:
Yes, this is an internal JSI API method that is expected to only be called by the end-user via `runOnUI` method (or some other methods that call to `runOnUI`) – the implementation of that method always converts the provided argument to "shareable" and if any other checks are necessary it'd make more sense to add them on the JS side in that runOnUI method. Otherwise this error is never expected to occur